listenbrainz/webserver/static/js/tests/user-settings/SelectTimezone.test.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import { mount, ReactWrapper } from "enzyme";
+import APIServiceClass from "../../src/utils/APIService";
+import GlobalAppContext from "../../src/utils/GlobalAppContext";
+
+import SelectTimezone from "../../src/user-settings/SelectTimezone";
+
+const user_timezone = 'America/New_York'
+const pg_timezones:Array<[string, string]> = [['Africa/Adnodjan', '+3:00:00 GMT'],['America/Adak', '-9:00:00 GMT']]
+
+const globalProps = {
+  APIService: new APIServiceClass(""),
+  currentUser: {
+    id: 1,
+    name: "testuser",
+    auth_token: "auth_token",
+  },
+  spotifyAuth: {},
+  youtubeAuth: {},
+};
+
+const props = {
+  pg_timezones,
+  user_timezone,
+  newAlert: () => {},
+};
+
+describe("submitTimezonePage", () => {
+
+  it("renders correctly", () => {
+    const extraProps = { ...props, newAlert: jest.fn() };
+    const wrapper = mount<SelectTimezone>(
+      <GlobalAppContext.Provider value={globalProps}>
+        <SelectTimezone {...extraProps} />
+      </GlobalAppContext.Provider>
+    );
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+});
+
+describe("resetTimezone", () => {
+  it("calls API, and sets state + creates a new alert on success", async () => {
```

Comment:
The description mentions it but this test doesn't have a assert that a alert is created. You can add it like this:

https://github.com/metabrainz/listenbrainz-server/blob/f2096ec1b80302a02fae65b551934c0e69cabf0a/listenbrainz/webserver/static/js/tests/pins/PinnedRecordingCard.test.tsx#L85-L99